dbt/adapters/athena/impl.py

```diff
@@ -28,18 +38,51 @@ def convert_text_type(cls, agate_table: agate.Table, col_idx: int) -> str:
         return "string"
 
     @classmethod
-    def convert_number_type(
-        cls, agate_table: agate.Table, col_idx: int
-    ) -> str:
+    def convert_number_type(cls, agate_table: agate.Table, col_idx: int) -> str:
         decimals = agate_table.aggregate(agate.MaxPrecision(col_idx))
         return "double" if decimals else "integer"
 
     @classmethod
-    def convert_datetime_type(
-            cls, agate_table: agate.Table, col_idx: int
-    ) -> str:
+    def convert_datetime_type(cls, agate_table: agate.Table, col_idx: int) -> str:
         return "timestamp"
 
+    def get_columns_in_relation(self, relation: Relation) -> List[Column]:
+        cached_relation = self.get_relation(relation.database, relation.schema, relation.identifier)
+        columns = []
+        if cached_relation and cached_relation.column_information:
+            for col_dict in cached_relation.column_information:
+                column = Column.create(col_dict["Name"], col_dict["Type"])
+                columns.append(column)
+            return columns
+        else:
+            return super().get_columns_in_relation(relation)
+
+    def list_relations_without_caching(self, schema_relation: AthenaRelation) -> List[AthenaRelation]:
+        relations = []
+        # Default quote policy of SQLAdapter
+        quote_policy = {"database": True, "schema": True, "identifier": True}
+        try:
+            for table in self._retrieve_glue_tables(schema_relation.database, schema_relation.schema):
+                rel_type = self._get_rel_type_from_glue_response(table)
+                relation = self.Relation.create(
+                    database=schema_relation.database,
+                    identifier=table["Name"],
+                    schema=schema_relation.schema,
+                    quote_policy=quote_policy,
+                    type=rel_type,
+                    column_information=table["StorageDescriptor"]["Columns"],
+                )
+                relations.append(relation)
+            return relations
+        except ClientError as e:
```

Comment:
When API call returns some error, Fallback into SQL implementation.
It cares some permission problems.